core/src/main/java/io/parsingdata/metal/token/Sub.java

```diff
@@ -32,6 +32,20 @@
 import io.parsingdata.metal.expression.value.OptionalValue;
 import io.parsingdata.metal.expression.value.ValueExpression;
 
+/**
+ * A {@link Token} that specifies a token to be parsed at a specific location
+ * in the input.
```

Comment:
Agreed. I prefer to call it input, but at some point we'll need to clarify somewhere that it can refer to different things...